extensions/map-layers-formats/src/Tools/GeometryTerrainDraper.ts

```diff
@@ -0,0 +1,189 @@
+    return status;
+  }
+
+  private rangeOverlapsLineString(range: Range3d) {
+    let inside = false;
+    const clipper = ConvexClipPlaneSet.createRange3dPlanes(range, true, true, true, true, false, false);
+    if (this._options.transform)
+      clipper.transformInPlace(this._options.transform);
+
+    for (let i = 0; i < this._points.length - 1 && !inside; i++)
+      inside = clipper.announceClippedSegmentIntervals(0, 1, this._points.getPoint3dAtUncheckedPointIndex(i), this._points.getPoint3dAtUncheckedPointIndex(i + 1));
+
+    return inside;
+  }
+}
+
+/** @internal */
+export class GeometryTerrainDraper implements TileUser {
+  // The side angle measures how close the sweep vector is from being parallel to a side face.
+  // The larger the angle, the more nearly vertical facets are ignored.
+  // This is an "empirical" value that was determined by looking at "problematic" polyfaces
+  public readonly sideAngle = Angle.createDegrees(0.06);
+
+  public readonly maxDistanceZ = 1.0E5; // Expand the Z Range, but not so much that we get opposite side of globe.
+  public readonly tileUserId: number;
+
+  public constructor(public readonly viewport: Viewport, public readonly treeRef: GeometryTileTreeReference) {
+    this.tileUserId = TileUser.generateId();
+    IModelApp.tileAdmin.registerUser(this);
+  }
+
+  public dispose(): void {
+    IModelApp.tileAdmin.forgetUser(this);
+  }
+
+  public get iModel() { return this.viewport.iModel; }
+
+  public onRequestStateChanged() {
+    this.viewport.invalidateDecorations();
+  }
+
+  public discloseTileTrees(trees: DisclosedTileTreeSet) {
+    trees.disclose(this.treeRef);
+  }
+
+  // Filter out non-top facets:
+  // For unknown reasons, there are "perpendicular" facets appearing in the terrain meshes.
+  public getMeshTopFacets(mesh: IndexedPolyface) {
+    // constant inputs
+    const sweepVector = Vector3d.unitZ();
+    // i.e., region is horizontal
+    // create subset visitor from the top facets
+    const topFacets: number[] = [];
+    const facetNormal = Vector3d.createZero();
+
+    for (const visitor = mesh.createVisitor(0); visitor.moveToNextFacet(); ) {
+      if (PolygonOps.unitNormal(visitor.point, facetNormal)) {
+        const theta = facetNormal.angleFromPerpendicular(sweepVector);
+        if (!theta.isMagnitudeLessThanOrEqual(this.sideAngle)) { // skip side facet
+          if (facetNormal.dotProduct(sweepVector) > 0)  // this is a top facet
+            topFacets.push(visitor.currentReadIndex());
+        }
+      }
+    }
+
+    return IndexedPolyfaceSubsetVisitor.createSubsetVisitor(mesh, topFacets, 0);
+  }
+
+  public drapeLineString(outStrings: CurvePrimitive[], inPoints: GrowableXYZArray, tolerance: number, range: Range3d): "loading" | "complete" {
+    const tree = this.treeRef.treeOwner.load();
+    if (!tree)
+      return "loading";
```

Comment:
nit: i hate the use of literal strings being used over and over in this file; would prefer pulling these into constants.

Reply:
ok will take note of that in future updates

Reply:
@aruniverse can you please resolve